### PR TITLE
HARMONY-1708: Update frontend and backend to be able to use TLS

### PR DESCRIPTION
**Pull request by chris-durbin**
## Jira Issue ID
HARMONY-1708

## Description
With the USE_HTTPS variable set to true both the harmony frontend and backend listeners will use TLS. We will set this to true in the deployed environments. See the ci-cd repo for instructions on how to test there.

## Local Test Steps
Verify that locally everything continues to work as expected by default (and does not use TLS). Verify in the sandbox that setting USE_HTTPS to true works correctly.

## PR Acceptance Checklist
* [X] Acceptance criteria met
* [ ] Tests added/updated (if needed) and passing
* [ ] Documentation updated (if needed)

### CODE REVIEW

File: services/harmony/app/server.ts [hunk 1]
```diff
@@ -91,7 +93,19 @@ function buildBackendServer(port: number, hostBinding: string): Server {
   app.get('/', ((req, res) => res.send('OK')));
   app.use(errorHandler);
 
-  return app.listen(port, hostBinding, () => appLogger.info(`Application backend listening on ${hostBinding} on port ${port}`));
+  let listener;
+  if (useHttps === 'true') {
```
Comment (indiejames) [inline services/harmony/app/server.ts:97]: For safety sake maybe change this to
```
if (useHttps.toLowerCase() === 'true') {
```

File: services/harmony/app/util/url.ts [hunk 2]
```diff
@@ -7,6 +7,9 @@ import * as url from 'url';
  * @returns The protocol (http or https) to use for public Harmony URLs
  */
 function _getProtocol(req): string {
+  if (process.env.USE_HTTPS === 'true') {
```
Comment (indiejames) [inline services/harmony/app/util/url.ts:10]: Same comment here about `toLowerCase`

File: services/harmony/app/server.ts [hunk 3]
```diff
@@ -130,7 +144,19 @@ function buildFrontendServer(port: number, hostBinding: string, config: RouterCo
   ogcCoveragesApi.handleOpenApiErrors(app);
   app.use(errorHandler);
 
-  return app.listen(port, hostBinding, () => appLogger.info(`Application frontend listening on ${hostBinding} on port ${port}`));
+  let listener;
+  if (config.USE_HTTPS === 'true') {
```
Comment (indiejames) [inline services/harmony/app/server.ts:148]: Same comment here about `toLowerCase`

File: services/harmony/app/server.ts (hunk 1 above)
Comment (indiejames) [reply services/harmony/app/server.ts:97]: Or maybe just convert it to lowercase when you first set up the config from the env var

**Review by indiejames — COMMENTED** [review]
Looks good. I will test then approve pending outcome.

**Review by indiejames — APPROVED** [review]
Tested locally

**Review by indiejames — APPROVED** [review]
tested locally

File: services/harmony/app/server.ts (hunk 1 above)
Comment (chris-durbin) [reply services/harmony/app/server.ts:97]: I prefer checking for the exact string match - this is a config parameter we directly control so we don't need to be flexible on what we support unlike something like a query parameter from a user.

**Review by ygliuvt — APPROVED** [review]

**Pull request merged**